### PR TITLE
Domains: Standardize domain mapping/transfer pages

### PR DESCRIPTION
**Pull request by sixhours**
These changes take effect when someone searches for a domain to map or transfer that has not been registered already, so we present them with the option to register it through us.

* Removes the Notice component; it's repetitive since we already note the domain in two other places (the search form and the domain search card) and global notices typically aren't used like this.
* Removes the green border around the search result.
* Adds a notice-like green border on the left side of the domain results to draw attention.

**Before**

screen shot 2018-08-27 at 12 17 33 pm

**After**

screen shot 2018-08-27 at 12 07 08 pm

**Steps to test**

1) Switch to this PR and navigate to `/start/`
2) At the domains step, choose to transfer or map a domain you already own.
3) Search for a domain name that has not been registered.
4) Note the design changes.

Ideally, I'd like to add "is available" after the domain name, like so, but I haven't figured out how to do it:

screen_shot_2018-07-27_at_3 56 52_pm

### CODE REVIEW

**Comment by klimeryk** [comment]
@sixhours Could you rebase this branch with the newest `master`, since it has conflicts right now? 🙏  I can help if you want.

**Comment by sixhours** [comment]
@klimeryk Oh, I thought I already resolved the conflicts in this commit: https://github.com/Automattic/wp-calypso/pull/26926/commits/3945835d7439ded07d9e086cd20a470e4877e2d6

But let me know if I need to go back and try again!

**Comment by klimeryk** [comment]
Unfortunately, looks like they were new, conflicting changes merged to `master` since then as well :( So it needs a rebase again to resolve the new conflicts (see the "This branch cannot be rebased due to conflicts" notice):
screen shot 2018-08-30 at 00 47 26

**Comment by sixhours** [comment]
@klimeryk This is weird, I'm seeing this:

screen shot 2018-08-29 at 7 19 00 pm

But I rebased locally and then committed a minor spacing change; maybe that's fixed it?

File: client/components/domains/map-domain-step/style.scss [hunk 1]
```diff
@@ -31,6 +31,10 @@
 	.domain-search-results__domain-availability {
 		margin-top: 30px;
 	}
+
+	.domain-suggestion.card {
+		border-left: 3px solid #4ab866;
```
Comment (jancavan) [inline client/components/domains/map-domain-step/style.scss]: We should use `$alert-green`.

File: client/components/domains/transfer-domain-step/style.scss [hunk 2]
```diff
@@ -37,6 +37,10 @@
 	.transfer-domain-step__domain-availability {
 		margin-top: 30px;
 	}
+
+	.domain-suggestion.card {
+		border-left: 3px solid #4ab866;
```
Comment (jancavan) [inline client/components/domains/transfer-domain-step/style.scss]: `$alert-green` here too.

File: client/components/domains/transfer-domain-step/index.jsx [hunk 3]
```diff
@@ -380,11 +380,6 @@ class TransferDomainStep extends React.Component {
 
 		return (
 			<div className={ 'transfer-domain-step__domain-availability' }>
-				<Notice status="is-success" showDismiss={ false }>
```
Comment (jancavan) [inline client/components/domains/transfer-domain-step/index.jsx]: Do we need the importing of this component too? Assuming it isn't being used elsewhere:

https://github.com/Automattic/wp-calypso/blob/199dbcce32577681ef7485c7236be1391632e35a/client/components/domains/transfer-domain-step/index.jsx#L32

Comment (sixhours) [reply client/components/domains/transfer-domain-step/index.jsx]: Yep, it's used for the notice that shows up when you try to use an illegal character in the domain field.

Comment (jancavan) [reply client/components/domains/transfer-domain-step/index.jsx]: Yeah, sorry, just noticed it now too. I got disconnected from proxy and some error message showed up using the component.

File: client/components/domains/transfer-domain-step/style.scss (hunk 2 above)
Comment (sixhours) [reply client/components/domains/transfer-domain-step/style.scss]: Oops, these should be fixed now. Thank you!

**Comment by jancavan** [comment]
> Ideally, I'd like to add "is available" after the domain name, like so, but I haven't figured out how to do it

I was also thinking of suggesting this before I read this part :) I think it'd be nice to add.

**Comment by sixhours** [comment]
I've added the availability note, but I'm not sure looking for a string within the current page path is the best way to determine which page we're on. I feel like there might be a less fragile way using `getSectionName` or something else? Needs a code review.

File: client/components/domains/domain-registration-suggestion/index.jsx [hunk 4]
```diff
@@ -119,8 +119,19 @@ class DomainRegistrationSuggestion extends React.Component {
...
-		return <h3 className="domain-registration-suggestion__title">{ domain }</h3>;
+
+		let isAvailable = false;
+
+		//If we're on the Mapping or Transfer pages, add a note about availability
+		if ( includes( page.current, '/mapping' ) || includes( page.current, '/transfer' ) ) {
+			isAvailable = true;
+		}
+
+		const title = isAvailable ? translate( '%s is available!', { args: domain } ) : domain;
```
Comment (a8ci18n) [inline client/components/domains/domain-registration-suggestion/index.jsx]: Hi! I've found a possible matching string that has already been translated **12** times:
`translate( '%s is available.' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/domains-transfer-standardization)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**